### PR TITLE
Pipeline: fix unnecessary warnings

### LINKED ISSUE #33498: The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
### System Info

- `transformers` version: 4.44.2
- Platform: Linux-6.8.0-44-generic-x86_64-with-glibc2.39
- Python version: 3.12.3
- Huggingface_hub version: 0.24.7
- Safetensors version: 0.4.5
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.4.1+cu121 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: No

### Who can help?

speech models: @ylacombe, @eustlb
pipelines: @Rocketknight1

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
import torch 
from transformers import pipeline

pipe = pipeline(
    "automatic-speech-recognition",
    model="openai/whisper-base.en",
    device="cpu",
    torch_dtype=torch.float32,
)

# https://github.com/openai/whisper/blob/main/tests/jfk.flac
pipe("./jfk.flac")
```

### Expected behavior

This does return the expected:

```python
{'text': ' And so my fellow Americans ask not what your country can do for you, ask what you can do for your country.'}
```

But it also prints the following, so would be nice to fix/suppress:

```
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
```

Thanks!

### PR DESCRIPTION
**Pull request by eustlb**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/issues/33498 https://github.com/huggingface/transformers/issues/35524

Handles two unnecessary warnings for the ASR pipeline:
- attention mask related: need to set `return_attention_mask=True` in `chunk_iter`. Its done anyway when not using `chunk_length_s` parameter ([here](https://github.com/huggingface/transformers/blob/099d93d2e9a1a5000f43f12fe63d66951f09a248/src/transformers/pipelines/automatic_speech_recognition.py#L445C12-L470C22))
- warning triggered when using deprecated input name for Whisper.

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
LGTM
